src/services/filter/filters/midnight/helpers.rs

```diff
@@ -3,3 +3,443 @@
 //! This module provides utility functions for working with Midnight-specific data types
 //! and formatting, including address normalization, value parsing, and
 //! operation processing.
+
+use crate::models::MidnightChainType;
+
+use midnight_ledger::{
+	base_crypto::hash::{HashOutput, PERSISTENT_HASH_BYTES},
+	serialize::deserialize,
+	storage::DefaultDB,
+	structure::{Proof, Proofish, Transaction as MidnightNodeTransaction, TransactionHash},
+	transient_crypto::encryption,
+	zswap::{
+		keys::{SecretKeys, Seed},
+		CoinCiphertext,
+	},
+};
+use midnight_node_ledger_helpers::{CoinInfo, NetworkId, DB};
+
+/// Parse a transaction index item
+pub fn parse_tx_index_item<P: Proofish<DefaultDB>>(
+	hash_without_prefix: &str,
+	raw_tx_data: &str,
+	network_id: NetworkId,
+) -> Result<
+	(
+		TransactionHash,
+		Option<MidnightNodeTransaction<P, DefaultDB>>,
+	),
+	anyhow::Error,
+> {
+	let hash = hex::decode(hash_without_prefix)
+		.map_err(|e| anyhow::anyhow!("TransactionHashDecodeError: {}", e))?;
+
+	// TODO: For now we always return early because there is an issue with deserialising tx data from raw_tx_data while the Midnight node team investigates
+	// Remove this once the issue is fixed
+	// TransactionDeserializeError: Invalid input data for core::option::Option<midnight_ledger::structure::Transaction<midnight_ledger::structure::Proof, midnight_storage::db::InMemoryDB>>,
+	//                              received version: None, maximum supported version is None. Invalid discriminant: 4
+	if !raw_tx_data.is_empty() {
+		return Ok((
+			TransactionHash(HashOutput(
+				hash.try_into()
+					.map_err(|_| anyhow::anyhow!("Invalid hash length"))?,
+			)),
+			None,
+		));
+	}
+
```

Comment:
We have this logic in place temporarily to bypass serialization issues with Midnight's testnet at the moment. Their node team is investigating, but to unblock, we are not attempting to deserialize anything from `transactions_index`.